Add Creator/Custom Design Items

otaviocc opened the pull request:
## Description

There are so many creators and custom designs to track and try that sometimes it's hard to keep track of them. A good place to store these information would be in this companion app.

## Implementation

The main types added by this commit are:

- `Design`: simple model with title, code, and description.
- `DesignRowView` and `DesignRowViewModel`: view and view model used to populate the table.
- `DesignFormView` and `DesignFormViewModel`: view and view model used to add a new design to the list.

This commit also touches the caching layer, adding designs as an optional array.

## Screenshots

![screenshots](https://user-images.githubusercontent.com/139272/82765444-6b4f0a80-9e17-11ea-88ed-2622273dce94.png)

## Demo

![output](https://user-images.githubusercontent.com/139272/82765453-79049000-9e17-11ea-8bfa-7f17ef0e5b04.gif)

In case of bugs, I can help [Blathers](https://animalcrossing.fandom.com/wiki/Blathers) on that.

Dimillian commented:
Hello, love the feature. But in the collection screen, this segmented control will crop the label on top. If you try on an iPhone 8 for example. Or a landscape iPad. Maybe it's not the right place to be or maybe we have to reword the top selector to something custom. I'll review the code shortly.

otaviocc commented:
> Hello, love the feature. But in the collection screen, this segmented control will crop the label on top. If you try on an iPhone 8 for example. Or a landscape iPad. Maybe it's not the right place to be or maybe we have to rework the top selector to something custom. I'll review the code shortly.

Oops :-) Do you have any suggestion for the ui/ux, @Dimillian? I can give it a try.

otaviocc commented:
> Hello, love the feature. But in the collection screen, this segmented control will crop the label on top. If you try on an iPhone 8 for example. Or a landscape iPad. Maybe it's not the right place to be or maybe we have to rework the top selector to something custom. I'll review the code shortly.

I have suggestion and will implement and propose it later tonight.

Dimillian commented:
Yes sorry didn't had time to think about it yet. Can't wait to see your proposition.

renaudjenny commented:
What about having a `More` tab which directs to a list of "other" available collections.
![Screenshot 2020-05-26 at 18 46 20](https://user-images.githubusercontent.com/2297556/82933159-6dcd7380-9f81-11ea-9921-b4b4a3e90d20.png)
![Screenshot 2020-05-26 at 18 46 54](https://user-images.githubusercontent.com/2297556/82933175-745beb00-9f81-11ea-9ad2-0b53fe7e5204.png)

Dimillian commented:
I like it. I would move critters into the more tab and leave the custom lists as it is. We already have a whole fritters section in the dashboard. And List have a subscribe call :p

otaviocc commented:
![new](https://user-images.githubusercontent.com/139272/83046195-f7934480-a046-11ea-909d-482d890fa9e2.png)

I've added the "More" tab and moved Critters and Designs there. Let me know what you think, @Dimillian, @renaudjenny.

Dimillian commented:
Awesome !!!! I guess this will land in master very soon :)

renaudjenny commented:
@otaviocc awesome! Can't wait to have this on the app.
I quickly checked the code, and it's looking good!

renaudjenny commented on ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionMoreDetailView.swift:
```diff
@@ -0,0 +1,62 @@
+        List {
+            CollectionMoreDetailView(viewModel: CollectionMoreDetailViewModel())
+        }
+        .previewLayout(.sizeThatFits)
```
That's smart. I've never think about this one!

Dimillian commented:
Making few fixes for localization and merging, great works!
